consumer/migration/hermes_migrator.go

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (C) 2022 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package migration
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/mysteriumnetwork/node/config"
+	"github.com/mysteriumnetwork/node/identity"
+	"github.com/mysteriumnetwork/node/identity/registry"
+	"github.com/mysteriumnetwork/node/session/pingpong"
+	"github.com/mysteriumnetwork/payments/crypto"
+	"github.com/rs/zerolog/log"
+)
+
+const oldBalanceMigrationMinimumMyst = 0.1
+
+var openChannelTimeout = time.Hour
+
+// HermesMigrator migrate identity from old hermes to new.
+// It opens a new channel for new Hermes and sends all MYST to a new payment channel.
+type HermesMigrator struct {
+	transactor          *registry.Transactor
+	addressProvider     registry.AddressProvider
+	hps                 pingpong.HermesPromiseSettler
+	hermesURLGetter     *pingpong.HermesURLGetter
+	hermesCallerFactory pingpong.HermesCallerFactory
+}
+
+// NewHermesMigrator create new HermesMigrator
+func NewHermesMigrator(
+	transactor *registry.Transactor,
+	addressProvider registry.AddressProvider,
+	hermesURLGetter *pingpong.HermesURLGetter,
+	hermesCallerFactory pingpong.HermesCallerFactory,
+	hps pingpong.HermesPromiseSettler,
+) *HermesMigrator {
+	return &HermesMigrator{
+		transactor:          transactor,
+		addressProvider:     addressProvider,
+		hermesURLGetter:     hermesURLGetter,
+		hermesCallerFactory: hermesCallerFactory,
+		hps:                 hps,
+	}
+}
+
+// HermesClient responses for receiving info from Hermes
+type HermesClient interface {
+	GetConsumerData(chainID int64, id string) (pingpong.HermesUserInfo, error)
+}
+
+// Start begins migration from old hermes to new
+func (m *HermesMigrator) Start(id string) error {
+	chainID := config.GetInt64(config.FlagChainID)
+
+	activeHermes, err := m.addressProvider.GetActiveHermes(chainID)
+	if err != nil {
+		return fmt.Errorf("could not get hermes address: %w", err)
+	}
+	knownHermeses, err := m.addressProvider.GetKnownHermeses(chainID)
+	if err != nil {
+		return fmt.Errorf("could not get hermes address: %w", err)
+	}
+	oldHermeses := getOldHermeses(knownHermeses, activeHermes)
+	if len(oldHermeses) != 1 {
+		return nil
+	}
+	oldHermes := oldHermeses[0]
```

Comment:
here I would loop all of them if its not too hard, usually there should only be one to migrate from anyway